tests/test_agpypeline.py

```diff
@@ -0,0 +1,487 @@
+    except RuntimeError as error:
+        assert str(error) == "The Algorithm class method perform_process() must be overridden by a derived class"
+
+
+def test_configuration():
+    """Tests agpypeline's configuration.Configuration() class by initializing configuration.Configuration() and
+     checking to make sure that all necessary parameters are contained"""
+    config = configuration.Configuration
+    for entry in ['transformer_version', 'transformer_description', 'transformer_name', 'transformer_sensor',
+                  'transformer_type', 'author_name', 'author_email', 'contributors', 'repository']:
+        assert hasattr(config, entry)
+        if entry == 'contributors':
+            assert getattr(config, entry) == []
+        else:
+            assert getattr(config, entry) is None
+
+
+def test_entrypoint_handle_error():
+    """Tests entrypoint's handle_error function by passing in an error code and message"""
+    entry = entrypoint.__internal__()
+    bad_result = entry.handle_error(None, None)
+    assert bad_result == {'error': 'An error has occurred with error code (-1)', 'code': -1}
+    result = entry.handle_error(0, "test message")
+    assert result == {'error': 'test message', 'code': 0}
+
+
+def test_entrypoint_load_metadata():
+    """Tests entrypoint's load_metadata function by seeing if the result is the same as a previously called valid
+    result"""
+    entry = entrypoint.__internal__()
+    orig = json.load(open("data/entrypoint_load_metadata_orig.json"))
+    bad_result = entry.load_metadata("")
+    assert bad_result == {'error': "Unable to load metadata file ''"}
+    assert entry.load_metadata("data/08f445ef-b8f9-421a-acf1-8b8c206c1bb8_metadata_cleaned.json") == orig
+
+
+def test_entrypoint_check_params_result_error():
+    """Tests entrypoint's check_params_result_error function by passing in both an entry dictionary and
+    a dictionary with a simple error message and code and checking for correct output"""
+    entry = entrypoint.__internal__()
+    result_none = entry.check_params_result_error({})
+    assert result_none is None
+    result_code = entry.check_params_result_error({'code': 0})
+    assert result_code == {'error': 'Error returned from get_transformer_params with code: 0', 'code': -104}
+    result_code_and_error = entry.check_params_result_error({'code': 0, 'error': 0})
+    assert result_code_and_error == {'error': 'An error has occurred with error code (-104)', 'code': -104}
+
+
+def test_entrypoint_check_retrieve_results_error():
+    """Tests entrypoint's check_retrieve_results_error function by passing in invalid (and soon valid) retrieve
+    values"""
+    entry = entrypoint.__internal__()
+    result_none = entry.check_retrieve_results_error(None)
+    assert result_none is None
+    result = entry.check_retrieve_results_error((3, 3))
+    assert result is None
+    # I think I would need an actual transformer_retrieve argument in order to fully test this
```

Comment:
Faking data is acceptable as long as it's meaningful (for example an integer and a string)